app/src/main/java/org/traccar/client/MainActivity.java

```diff
@@ -307,4 +289,38 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
         }
     }
 
+    private boolean validateServerURL(String userUrl) {
+        if (userUrl == null || userUrl.trim().length() == 0) {
```

Comment:
I don't think we should explicitly handle this case. I feel like you are trying to overcomplicate it for no reason. There can never be null value as far as I understand it. If the value is empty, URL validation will handle it.